modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1185,6 +1185,12 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
         else if (layer_type == "Conv")
         {
             CV_Assert(node_proto.input_size() >= 2);
+            if (layerParams.has("kernel_size")) {
+                const DictValue &kernel = layerParams.get("kernel_size");
+                if (kernel.size() == 1) {
+                    layerParams.set("is_1d", true);
```

Comment:
It's interesting if we can avoid this flag. Can you please check if it's critical to have 2 parameters for 1D convolution? In example, we have kernel_size = (3, 3). Because we can determine 1D convolution not by `kernel_size.size() == 1` but by `inp.dims == 3`

Reply:
> It's interesting if we can avoid this flag. Can you please check if it's critical to have 2 parameters for 1D convolution? In example, we have kernel_size = (3, 3). Because we can determine 1D convolution not by `kernel_size.size() == 1` but by `inp.dims == 3`

I removed the flag and changed the logic for isConv*D variables. Please, check